src/components/sidebar-content-box/SidebarContentBox.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { FC, isValidElement, cloneElement, Children } from 'react'
+import { useTranslation } from 'react-i18next'
+import Box from '@mui/material/Box'
+import Typography from '@mui/material/Typography'
+
+import AppButton from '~/components/app-button/AppButton'
+import AllContentModal from '~/components/all-content-modal/AllContentModal'
+import FileComponent from '~/components/file-component/FileComponent'
+import LinkComponent from '~/components/link-component/LinkComponent'
+import SidebarImageGrid from '~/components/sidebar-image-grid/SidebarImageGrid'
+
+import { maxElemToShow } from '~/components/sidebar-content-box/SidebarContentBox.constants'
+import { useModalContext } from '~/context/modal-context'
+import { spliceSx } from '~/utils/helper-functions'
+import { SizeEnum, ButtonVariantEnum, Link, File } from '~/types'
+import { styles } from '~/components/sidebar-content-box/SidebarContentBox.styles'
+
+interface SidebarContentBoxProps {
+  icon: React.ComponentType
+  name: string
+  content?: Link[] | File[]
+}
+
+type ModifiedChildren = {
+  compactMode: boolean
+}
+
+const SidebarContentBox: FC<SidebarContentBoxProps> = ({
+  icon: Icon,
+  name,
+  content,
+  children
+}) => {
+  const { t } = useTranslation()
+  const { openModal } = useModalContext()
+
+  const seeAll = () => {
+    const updatedChildren = Children.map(children, (child) => {
+      if (isValidElement(child) && child.type === SidebarImageGrid) {
+        return cloneElement(child, { compactMode: false } as ModifiedChildren)
```

Comment:
Can you explain why you need to clone children?

Reply:
> Can you explain why you need to clone children?

To change received SidebarImageGrid child props, namely compactMode. That allows to render image grid fully in "see all" modal.

Reply:
You don't need to do if you make map for the received data in the parent component and after that pass it to this component as a prop